plugins/language_typst.lua

```diff
@@ -0,0 +1,157 @@
+        ["Iota"]     = "keyword2",
+        ["Kappa"]    = "keyword2",
+        ["Lambda"]   = "keyword2",
+        ["Mu"]       = "keyword2",
+        ["Nu"]       = "keyword2",
+        ["Xi"]       = "keyword2",
+        ["Omicron"]  = "keyword2",
+        ["Pi"]       = "keyword2",
+        ["Rho"]      = "keyword2",
+        ["Sigma"]    = "keyword2",
+        ["Tau"]      = "keyword2",
+        ["Upsilon"]  = "keyword2",
+        ["Phi"]      = "keyword2",
+        ["Chi"]      = "keyword2",
+        ["Psi"]      = "keyword2",
+        ["Omega"]    = "keyword2"
+    }
+}
+
+
+local typst_script = {
+
+    patterns = {
+
+        { pattern = { "/%*", "%*/" },              type = "comment" },
+        { pattern = "//.*",                        type = "comment" },
+
+        { pattern = { '"', '"', '\\' },            type = "string"  },
+        { pattern = {"%[", "%]"},                  type = "normal", syntax = ".typ" },
+
+        { pattern = "[%a_][%w_%-]*%f[(]",          type = "function" },
+
+        { pattern = "0x[%dabcdef]+",               type = "number" },
+        { pattern = "0b[01]+",                     type = "number" },
+        { pattern = "0o[01234567]+",               type = "number" },
+        { pattern = "%d+[.e]%d+()[%a%%]+",         type = {"number", "keyword2"}},
+        { pattern = "%d+[.e]%d+",                  type = "number" },
+        { pattern = "%d+()[%a%%]+",                type = {"number", "keyword2"}},
+        { pattern = "%d",                          type = "number" },
+
+
+        { pattern = "[%+%-=/%*%^%%<>~|&#_^\\]",    type = "operator" },
+
+    },
+
+
+    symbols = {
+        ["set"] = "keyword",
+        ["let"] = "keyword",
+        ["include"] = "keyword",
+    }
+}
+
+
+syntax.add {
+    name = "Typst",
+    files = { "%.typ$" },
+    comment = "//",
+    block_comment = { "/*", "*/" },
+
+    patterns = {
+
+
+        { pattern = {"%$", "%$", "\\"},            type = "literal", syntax = typst_math },
+        { pattern = {"```js", "```", "\\"},        type = "literal", syntax = ".js" },
+        { pattern = {"#", "[^,%(%[{]\n"},          type = "literal", syntax = typst_script },
+        { pattern = {"`", "`", "\\"},              type = "normal"},
+
+        { pattern = { '"', '"', '\\' },            type = "string"  },
+
+
+        { pattern = "//.*",                        type = "comment" },
+        { pattern = { "/%*", "%*/" },              type = "comment" },
+
+   --     { pattern = {"link"},                    type = "typst_underline"}, -- add underline text later
+
+        { pattern = "^=+ ().+%f[\n]",              type ={"operator", "normal"}}, -- Supposed to be bold
+
+    -- Finish later
+    --    { pattern = {"%*_", "_%*[%s,%.]"},     type = "bold_italic"},
+    --    { pattern = {"_%*", "%*_[%s,%.]"},     type = "bold_italic"},
+    --    { pattern = {"_", "_[%s,%.]"},         type = "italic"},
+    --    { pattern = {"%*[^%/]", "%*[%s,%.]"},  type = "bold"},
```

Comment:
Instead of leaving them out, what do you think about at least highlighting them?
Feel free to use any token type that looks fine.